src/com/goide/psi/impl/GoTypeUtil.java

```diff
@@ -43,4 +53,202 @@ public static boolean isString(@Nullable GoType type) {
+        result.add(GoElementFactory.createType(parent.getProject(), "chan " + getType(type, parent).getText()));
+      }
+      else {
+        result.add(getType(null, parent));
+      }
+    }
+
+    if (parent instanceof GoSendStatement || parent instanceof GoLeftHandExprList && parent.getParent() instanceof GoSendStatement) {
+      GoSendStatement sendStatement = (GoSendStatement)(parent instanceof GoSendStatement ? parent : parent.getParent());
+      GoLeftHandExprList leftHandExprList = sendStatement.getLeftHandExprList();
+      GoExpression channel =
+        ContainerUtil.getFirstItem(leftHandExprList != null ? leftHandExprList.getExpressionList() : sendStatement.getExpressionList());
+      GoExpression sendExpr = sendStatement.getSendExpression();
+      assert channel != null;
+      if (expression.isEquivalentTo(sendExpr)) {
+        GoType chanType = channel.getGoType(null);
+        if (chanType instanceof GoChannelType) {
+          result.add(getType(((GoChannelType)chanType).getType(), parent));
+          return result;
+        }
+      }
+
+      if (expression.isEquivalentTo(channel)) {
+        GoType type = sendExpr != null ? sendExpr.getGoType(null) : null;
+        result.add(GoElementFactory.createType(parent.getProject(), "chan " + getType(type, parent).getText()));
+        return result;
+      }
+
+      result.add(getType(null, parent));
+      return result;
+    }
+
+    if (parent instanceof GoExprCaseClause) {
+      GoExprSwitchStatement switchStatement = PsiTreeUtil.getParentOfType(parent, GoExprSwitchStatement.class);
+      if (switchStatement == null) return result;
+
+      GoType type;
+      GoExpression switchExpr = switchStatement.getExpression();
+      if (switchExpr != null) {
+        type = getType(switchExpr, parent);
+      }
+      else {
+        GoStatement statement = switchStatement.getStatement();
+        if (statement == null) {
+          type = GoElementFactory.createType(parent.getProject(), "bool");
+        }
+        else {
+          GoLeftHandExprList leftHandExprList = ((GoSimpleStatement)statement).getLeftHandExprList();
+          GoExpression expr = leftHandExprList != null ? ContainerUtil.getFirstItem(leftHandExprList.getExpressionList()) : null;
+          type = getType(expr, parent);
+        }
+      }
+      result.add(type);
+    }
+
+    return result;
+  }
+
+  @NotNull
+  private static GoType createGoTypeListOrGoType(@NotNull List<GoType> types, @NotNull PsiElement context) {
+    if (types.size() < 2) {
+      return getType(ContainerUtil.getFirstItem(types), context);
+    }
+    return GoElementFactory.createTypeList(context.getProject(), StringUtil.join(types, new Function<GoType, String>() {
+      @Override
+      public String fun(GoType type) {
+        return type == null ? "interface{}" : type.getText();
+      }
+    }, ", "));
+  }
+
+  @NotNull
+  private static GoType getType(@Nullable PsiElement element, @NotNull PsiElement context) {
```

Comment:
I cannot understand what this method does from name and what should I pass to it